packages/astro/src/actions/runtime/middleware.ts

```diff
@@ -0,0 +1,43 @@
+import { defineMiddleware } from '../../core/middleware/index.js';
+import { ApiContextStorage } from './store.js';
+import { formContentTypes, getAction } from './utils.js';
+import { callSafely } from './virtual/shared.js';
+import type { APIContext } from '../../@types/astro.js';
+
+export type Locals = {
+	_actionsInternal: {
+		getActionResult: APIContext['getActionResult'];
+	};
+};
+
+export const onRequest = defineMiddleware(async (context, next) => {
+	const locals = context.locals as Locals;
+	const { request } = context;
+	const contentType = request.headers.get('Content-Type');
+	locals._actionsInternal = {
+		getActionResult: () => undefined,
+	};
+
+	if (!formContentTypes.some((f) => contentType?.startsWith(f))) return next();
```

Comment:
You have to `.toLowerCase` the value of the headers. The values aren't case-sensitive. We do this in the check origin too. E.g. you might get `APPLICATION/JSON` as header value, and it's valid.

Also, why do we use`startsWith`?

Reply:
Good point! definitely need a `.toLowerCase` check. I also found the need for `startsWith` when using web-standard form POST requests. For whatever reason, Chrome may add a suffix to urlencoded forms like `urlencoded---other-info`. The `startsWith` parses correctly. Happy to add a comment explaining

Reply:
Oh really? That's... new to me 🤔 maybe it's worth leaving a comment for posterities

Reply:
Ah ha, this is what I was seeing. Chrome may apply a `boundary` property to the content-type. We could split on `;` for completeness from reading the spec
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type#content-type_in_html_forms